sql/core/src/test/scala/org/apache/spark/sql/execution/debug/DebuggingSuite.scala

```diff
@@ -149,3 +151,8 @@ class DebuggingSuite extends DebuggingSuiteBase with DisableAdaptiveExecutionSui
 }
 
 class DebuggingSuiteAE extends DebuggingSuiteBase with EnableAdaptiveExecutionSuite
+
+class DebuggingSuiteWithPartitionEvaluator extends DebuggingSuiteBase {
```

Comment:
can you measure the test time? If it's too costly we can skip testing it.

Reply:
After more thought, it seems confusing if we have partition evaluator tests here and there randomly. Maybe it's better to make a rule that we don't add extra tests for partition evaluators.

Reply:
The evaluator is actually tested anyway, and we can test `mapPartitionsWithEvaluator` after we decide to use it by default.